pyClient/zeth/signing.py

```diff
@@ -1,39 +1,111 @@
 """
-Types and basic operations for signing joinsplit data.
+Implementation of Schnorr-based one-time signature from: "Two-tier
+signatures, strongly unforgeable signatures, and Fiat-Shamir without random
+oracles" by Bellare and Shoup (https://eprint.iacr.org/2007/273.pdf) over Curve
+BN128
 """
 
-import zeth.constants as constants
+from math import ceil
+from os import urandom
+from hashlib import sha256
 from py_ecc import bn128 as ec
-from Crypto import Random
-from typing import Tuple, NewType
-
-
-FQ = ec.FQ
-G1 = Tuple[ec.FQ, ec.FQ]
+from zeth.utils import FQ, G1, encode_g1_to_bytes
+import zeth.constants as constants
```

Comment:
we only use `ZETH_PRIME`, it would be more efficient to do:
`from zeth.constants import ZETH_PRIME`

Reply:
Updated